graph/api/api_test.go

```diff
@@ -3284,13 +3284,13 @@ func TestComplexGraph(t *testing.T) {
 	mockQuery(xapi, q16, &v16)
 	mockQuery(xapi, q17, &v17)
 
-	var fut func(b *business.Layer, p *prometheus.Client, o graph.Options) (int, interface{})
+	var fut func(b *business.Layer, p *prometheus.Client, o graph.Options, ctx context.Context) (int, interface{})
 
 	mr := mux.NewRouter()
 	mr.HandleFunc("/api/namespaces/graph", http.HandlerFunc(
 		func(w http.ResponseWriter, r *http.Request) {
 			context := context.WithValue(r.Context(), "authInfo", &api.AuthInfo{Token: "test"})
-			code, config := fut(nil, client, graph.NewOptions(r.WithContext(context)))
+			code, config := fut(nil, client, graph.NewOptions(r.WithContext(context)), context)
```

Comment:
Is there an easy way to test with some non-null health data?

Reply:
I think it is testable.
I would need to pass the `includeHealth` flag set as true and a non-nil business layer mock.

Then, the tedious part would be to mock all calls that are being done to the cluster API and prom :-/
I'll check how bad it is.